mlflow/xgboost/__init__.py

```diff
@@ -722,6 +723,47 @@ def log_feature_importance_plot(features, importance, importance_type):
             # training model
             model = original(*args, **kwargs)
 
+            # dtrain must exist as the original train function already ran successfully
+            # it is possible that the dataset was constructed before the patched
+            #   constructor was applied, so we cannot assume the input_example_info exists
+            input_example_info = getattr(dtrain, "input_example_info", None)
+
+            def get_input_example():
+                if input_example_info is None:
+                    raise Exception(ENSURE_AUTOLOGGING_ENABLED_TEXT)
+                if input_example_info.error_msg is not None:
+                    raise Exception(input_example_info.error_msg)
+                return input_example_info.input_example
+
+            def infer_model_signature(input_example):
+                model_output = model.predict(xgboost.DMatrix(input_example))
+                return infer_signature(input_example, model_output)
+
+            # Only log the model if the autolog() param log_models is set to True.
+            model_id = None
+            if _log_models:
+                # Will only resolve `input_example` and `signature` if `log_models` is `True`.
+                input_example, signature = resolve_input_example_and_signature(
+                    get_input_example,
+                    infer_model_signature,
+                    log_input_examples,
+                    log_model_signatures,
+                    _logger,
+                )
+
+                registered_model_name = get_autologging_config(
+                    FLAVOR_NAME, "registered_model_name", None
+                )
+                logged_model = log_model(
```

Comment:
Shouldn't we include the params as well?

Reply:
Great catch, added!